lib/src/checkbox/Checkbox.tsx

```diff
@@ -24,92 +27,82 @@ const DxcCheckbox = ({
   size = "fitContent",
   tabIndex = 0,
 }: CheckboxPropsType): JSX.Element => {
-  const [switchId] = useState(`checkbox-${uuidv4()}`);
-  const labelId = `label-${switchId}`;
+  const labelId = `label-checkbox` + useId();
   const [innerChecked, setInnerChecked] = useState(defaultChecked);
-  const [isLabelHovered, setIsLabelHovered] = useState(false);
+
+  const checkboxRef = useRef(null);
 
   const colorsTheme = useTheme();
   const backgroundType = useContext(BackgroundColorContext);
   const translatedLabels = useTranslatedLabels();
 
-  const handlerCheckboxChange = (checkboxValue) => {
-    if (checked === undefined) {
-      const isChecked = checkboxValue.target.checked === undefined ? !innerChecked : checkboxValue.target.checked;
-      setInnerChecked(isChecked);
-      if (typeof onChange === "function") {
-        onChange(isChecked);
-      }
-    } else {
-      if (typeof onChange === "function") {
-        onChange(!checked);
-      }
+  const handleCheckboxChange = () => {
+    if (document.activeElement !== checkboxRef?.current) {
+      checkboxRef?.current.focus();
```

Comment:
We need to also check the nullish value `current`: `checkboxRef?.current?.focus();`